org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/DetectVMInstallationsJob.java

```diff
@@ -48,12 +48,17 @@ public class DetectVMInstallationsJob extends Job {
 
 	private static final Object FAMILY = DetectVMInstallationsJob.class;
 
+	private static final String CI_ENV_VARIABLE = System.getProperty("jdt.jvmdetect.envCI", "CI"); //$NON-NLS-1$ //$NON-NLS-2$
+
 	public DetectVMInstallationsJob() {
 		super(LaunchingMessages.lookupInstalledJVMs);
 	}
 
 	@Override
 	public IStatus run(IProgressMonitor monitor) {
+		if (!CI_ENV_VARIABLE.isEmpty() && Boolean.parseBoolean(System.getenv(CI_ENV_VARIABLE))) {
```

Comment:
This would be a bit simpler and should handle cases where `jdt.jvmdetect.envCI` is set to an empty string or alike, by simply returning `false`.
```suggestion
		if (Boolean.getBoolean(CI_ENV_VARIABLE)) {
```

Reply:
Get boolean returns a system property CI is an environment variable.